__tests__/__unit__/DatasetTree.unit.test.ts

```diff
@@ -717,6 +719,36 @@ describe("DatasetTree Unit Tests", () => {
         expect(testTree.mFavorites[0].label).toBe(`[${sessionNode.label.trim()}]: ${newLabel}`);
     });
 
+    it("Should rename a dataset", async () => {
```

Comment:
I would suggest to split this test on 5 separate ones with clear description of what you test when change context_value.
If you need to test shift from one context value to another it should be again separate test.

It will also eliminate necessity to check number of calls to **rename** function.